src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -0,0 +1,436 @@
+
+
+def _blob_is_not_complete(metadata):
+    if not metadata:
+        return True
+
+    for key in metadata:
+        if key.lower() == 'complete':
+            return False
+
+    return True
+
+
+def _get_blob_info(blob_sas_url):
+    match = re.search((r"http(s)?://(?P<account_name>.*?)\.blob\.(?P<endpoint_suffix>.*?)/(?P<container_name>.*?)/"
+                       r"(?P<blob_name>.*?)\?(?P<sas_token>.*)"), blob_sas_url)
+    account_name = match.group('account_name')
+    endpoint_suffix = match.group('endpoint_suffix')
+    container_name = match.group('container_name')
+    blob_name = match.group('blob_name')
+    sas_token = match.group('sas_token')
+
+    if not account_name or not container_name or not blob_name or not sas_token:
+        raise CLIError("Failed to parse the SAS URL: '{!s}'.".format(blob_sas_url))
+
+    return account_name, endpoint_suffix, container_name, blob_name, sas_token
+
+
+def acr_build(cmd,
+              client,
+              registry_name,
+              source_location,
+              image_names=None,
+              resource_group_name=None,
+              timeout=None,
+              build_arg=None,
+              secret_build_arg=None,
+              docker_file_path='Dockerfile',
+              no_logs=False):
+    _, resource_group_name = validate_managed_registry(
+        cmd.cli_ctx, registry_name, resource_group_name, BUILD_NOT_SUPPORTED)
+
+    client_registries = cf_acr_registries(cmd.cli_ctx)
+
+    tar_file_path = os.path.join(tempfile.gettempdir(), 'source_archive_{}.tar.gz'.format(hash(os.times())))
+
+    if os.path.exists(source_location):
+        if not os.path.isdir(source_location):
+            raise CLIError("'--context -c' should be a local directory path or remote URL.")
+
+        _check_local_docker_file(source_location, docker_file_path)
+        size = 0
+
+        try:
+            source_location = _upload_source_code(
+                client_registries, registry_name, resource_group_name, source_location, tar_file_path, docker_file_path)
+            size = os.path.getsize(tar_file_path)
+        except Exception as err:
+            raise CLIError(err)
+        finally:
+            try:
+                logger.debug("Starting to delete the archived source code from '%s'.", tar_file_path)
+                os.remove(tar_file_path)
+            except OSError:
+                pass
+
+        unit = 'GiB'
+        for S in ['Bytes', 'KiB', 'MiB', 'GiB']:
+            if size < 1024:
+                unit = S
+                break
+            size = size / 1024.0
+        print("Sending build context ({0:.3f} {1}) to ACR.".format(size, unit))
+    else:
+        source_location = _check_remote_source_code(source_location)
+        print("Sending build context to ACR.")
+
+    if image_names:
+        is_push_enabled = True
+    else:
+        is_push_enabled = False
+        print("'--image -t' is not provided. Skip image push after build.")
```

Comment:
Use `logger.warning` for these user style messages.

Reply:
Same with below on messages like 'Waiting for a build agent...' etc.

By using logger, the messages go to stderr instead of stdout.